sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/DocumentsImpl.java

```diff
@@ -57,8 +59,11 @@
      *
      * @param client the instance of the service client containing this operation class.
      */
-    DocumentsImpl(SearchIndexClientImpl client) {
-        this.service = RestProxy.create(DocumentsService.class, client.getHttpPipeline());
+    DocumentsImpl(SearchIndexClientImpl client, SerializerAdapter serializerAdapter) {
+        if (serializerAdapter == null) {
+            serializerAdapter = new JacksonAdapter();
+        }
+        this.service = RestProxy.create(DocumentsService.class, client.getHttpPipeline(), serializerAdapter);
```

Comment:
@srnagar, looks like this will require codegen changes.

Reply:
This is a POC of whether Search can adopt the autorest chagnes. Will regen the code after autorest change get merged.